test/marketplace/MarketPlace.sol

```diff
@@ -49,30 +49,29 @@ contract MarketPlace {
     swivel = s;
     return true;
   }
-
+  
   /// @notice Allows the owner to create new markets
-  /// @param u Underlying token address associated with the new market
   /// @param m Maturity timestamp of the new market
   /// @param c cToken address associated with underlying for the new market
   /// @param n Name of the new zcToken market
   /// @param s Symbol of the new zcToken market
-  function createMarket(
-    address u,
+   function createMarket(
     uint256 m,
     address c,
     string memory n,
     string memory s,
     uint8 d
   ) public onlyAddress(admin) returns (bool) {
     address swivelAddr = swivel;
-    require(markets[u][m].vaultAddr == address(0), 'market already exists');
+    address uAddr = CErc20(c).underlying(); 
+    require(markets[uAddr][m].vaultAddr == address(0), 'market already exists');
     require(swivelAddr != address(0), 'swivel contract address not set');
     // TODO can we live with the factory pattern here both bytecode size wise and CREATE opcode cost wise?
-    address zctAddr = address(new ZcToken(u, m, n, s, d));
+    address zctAddr = address(new ZcToken(uAddr, m, n, s, d));
     address vAddr = address(new VaultTracker(m, c, swivelAddr));
-    markets[u][m] = Market(c, zctAddr, vAddr, 0);
```

Comment:
get `u` from the ctoken. is this worth the call? 

is the invariant -> `u` == ctoken.underlying true?

if so i suppose it could be a potential source of error, that the user passed an underlying that was different from the ctoken one. yeah, ok if so.

Reply:
Yeah since were the only ones creating markets / even if we make it an open protocol in the future, itll be important to make sure an invalid market cant be created.

This essentially just makes sure that the entry point is an existing cToken market to use, and from there gets the uAddr and decimals

Reply:
done.